src/goCheck.ts

```diff
@@ -160,9 +165,19 @@ export function check(filename: string, goConfig: vscode.WorkspaceConfiguration)
 				if (!isMainPkg) {
 					args.push('-i');
 				};
-				args = args.concat(['-o', tmppath, '-tags', buildTags, ...buildFlags, '.']);
+				args = args.concat(['-tags', buildTags, ...buildFlags]);
 				if (filename.match(/_test.go$/i)) {
-					args = ['test', '-copybinary', '-o', tmppath, '-c', '-tags', buildTags, ...buildFlags, '.'];
+					args = ['test', '-tags', buildTags, ...buildFlags];
+					if (multiPackage) {
+						args = args.concat('-run=^$');
+					} else {
+						args = args.concat('-copybinary', '-c');
+					}
+				}
+				if (multiPackage) {
+					args = args.concat('./...');
```

Comment:
coverOnSave is broken when multiPackage is set to true
To begin with, we can skip test from the multiPackage feature, and continue working on it in a separate PR

https://github.com/Microsoft/vscode-go/issues/816 talks about using test results from multiple packages for getting code coverage. We can use that to track multiPackage feature for test and code coverage